applications/serial_lte_modem/src/slm_at_tcpip.c

```diff
@@ -792,6 +802,116 @@ int handle_at_socket(enum at_cmd_type cmd_type)
 	return err;
 }
 
+/**@brief handle AT#XSOCKET commands
+ *  AT#XSSOCKET=<op>[,<type>,<role>,sec_tag>[,<peer_verify>[,<hostname_verify>]]]
+ *  AT#XSSOCKET?
+ *  AT#XSSOCKET=?
+ */
+int handle_at_secure_socket(enum at_cmd_type cmd_type)
+{
+	int err = -EINVAL;
+	uint16_t op;
+
+	switch (cmd_type) {
+	case AT_CMD_TYPE_SET_COMMAND:
+		err = at_params_unsigned_short_get(&at_param_list, 1, &op);
+		if (err) {
+			return err;
+		}
+		if (op == AT_SOCKET_OPEN) {
+			uint16_t type;
+			uint16_t role;
+			/** Peer verification level for TLS connection.
+			 *    - 0 - none
+			 *    - 1 - optional
+			 *    - 2 - required
+			 * If not set, socket will use defaults (none for servers,
+			 * required for clients)
+			 */
+			uint16_t peer_verify = 2;
```

Comment:
nit, but you could use symbols from net/socket.h: `TLS_PEER_VERIFY_REQUIRED/OPTIONAL/NONE`, same below.